### ISSUE TITLE
Installing Node failed

### ISSUE THREAD

**Issue by qido2** [screepers/screeps-launcher#40]
We are trying to set up a private server on Raspberry Pi 3 running Raspbian following this guide using the latest version:
- [Newbie-friendly (ish) private/dedicated server setup guide for Ubuntu 18.04, with automatic startup](https://www.reddit.com/r/screeps/comments/deyq66/newbiefriendly_ish_privatededicated_server_setup/)

Once we try to run the `./screeps-launcher` inside terminal it tries to install NodeJS and fails with error code `404`. 

![image](https://github.com/screepers/screeps-launcher/assets/90379047/fa5e5403-4bf4-4e6d-94de-1bd3c0786360)

We are using the [screeps-launcher_v1.15.1_linux_arm](https://github.com/screepers/screeps-launcher/releases/download/v1.15.1/screeps-launcher_v1.15.1_linux_arm) version as it is the only we were actually able to execute. As we later found out inside the source code of the [install.go](https://github.com/screepers/screeps-launcher/blob/d0b47e679c9639896dce7580a144c87f0ba1e6cf/install/install.go#L201C18-L201C18) file at line [201](https://github.com/screepers/screeps-launcher/blob/d0b47e679c9639896dce7580a144c87f0ba1e6cf/install/install.go#L201), it sets the `arch` to `armv6l` and tries to install NodeJS based on that.

We then found out there is no such distributable for the NodeJS based on the [Index of /dist/v12.22.12/](https://nodejs.org/dist/v12.22.12/) and our Raspberry is also running on `armv7l` architecture as shown in the screenshot below.

![image](https://github.com/screepers/screeps-launcher/assets/90379047/93514a3e-a74f-41de-b840-3a66f2c735d8)

We are able to manually install the correct `armv7l` version of NodeJS, but the `./screeps-launcher` keeps looking for the `armv6l` version as it is predefined in the code.

Any ideas on how to tackle this?

**Comment by AlinaNova21** [comment]
Haven't messed with that recently, but may want to try a 64 bit os and arm 64, you can also extract a node tar into the deps/node folder to manually install it. I'll look into setting what it would take to fix this issue at some point, although in my experience a pi really doesn't like the server

**Comment by qido2** [comment]
> Haven't messed with that recently, but may want to try a 64 bit os and arm 64, you can also extract a node tar into the deps/node folder to manually install it. I'll look into setting what it would take to fix this issue at some point, although in my experience a pi really doesn't like the server

We are trying to fiddle with it a bit more. I will keep this post updated with what we find.

**Comment by ckreon** [comment]
I have a similar issue when installing on Ubuntu 18.04 LTS - I can get all the way to running screeps-launcher, but when it goes to install Node I get this error:

```
2023/09/16 16:16:32 screeps-launcher v1.15.1 (Wed Sep 13 22:01:35 UTC 2023)
2023/09/16 16:16:38 Loaded config from config.yml
2023/09/16 16:16:38 Installing Node
2023/09/16 16:16:40 Get https://nodejs.org/dist/index.json: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

Not sure exactly what is causing this at the moment, I've poked into it a bit, but the usual suspects aren't the issue (/etc/hosts, proxies, and so forth).

**Comment by BlueskyFR** [comment]
Same issue here on standard Linux using the docker image

**Comment by BlueskyFR** [comment]
Ok so on my side I was using `podman-compose` so since root was owning the `/screeps` directory and the default `screeps` user has no permissions I had to add `user: root` to my `docker-compose.yml`!

**Comment by qido2** [comment]
We were unable to find any solution.

We've tried to:
1. Clean install of the newest Raspbian version for the Raspberry Pi 3.
2. Manually set the `arch` value to `armv7l` in the source code and recompile the launcher.
    - this fixed the `404` error message when installing NodeJS.
3. After fixing the  `404` error message we encountered a few missing packages and were able to install them manually.
4. What followed was a never-ending battle with `yarn` and package versions.

In conclusion, we gave up for now, as we don't have enough spare time to continue this investigation.
